src/services/stringCompletions.ts

```diff
@@ -247,16 +247,17 @@ namespace ts.Completions.StringCompletions {
         const scriptPath = sourceFile.path;
         const scriptDirectory = getDirectoryPath(scriptPath);
 
-        const extensionOptions = getExtensionOptions(compilerOptions);
-        if (isPathRelativeToScript(literalValue) || isRootedDiskPath(literalValue)) {
-            if (compilerOptions.rootDirs) {
-                return getCompletionEntriesForDirectoryFragmentWithRootDirs(
-                    compilerOptions.rootDirs, literalValue, scriptDirectory, extensionOptions, compilerOptions, host, scriptPath);
+        if (isRootedDiskPath(literalValue) || isUrl(literalValue)) {
+            if (compilerOptions.baseUrl) {
+                return getCompletionEntriesForNonRelativeModules(literalValue, scriptDirectory, compilerOptions, host, typeChecker);

```

Comment:
The two `getCompletionEntriesForNonRelativeModules` calls are identical, as are the two `getCompletionEntriesForRelativeModules` calls. This could be a single conditional instead of several nested ones.
```ts
return isPathRelativeToScript(literalValue) || !compilerOptions.baseUrl && (isRootedDiskPath(literalValue) || isUrl(literalValue))
    ? getCompletionEntriesForRelativeModules(literalValue, scriptDirectory, compilerOptions, host, scriptPath)
    : getCompletionEntriesForNonRelativeModules(literalValue, scriptDirectory, compilerOptions, host, typeChecker);
```